CMakeLists.txt

```diff
@@ -287,6 +288,8 @@ set_target_properties(rocksdb${ARTIFACT_SUFFIX} PROPERTIES COMPILE_FLAGS "-DROCK
 add_dependencies(rocksdb${ARTIFACT_SUFFIX} GenerateBuildVersion)
 target_link_libraries(rocksdb${ARTIFACT_SUFFIX} ${LIBS})
 
+include(${CMAKE_CURRENT_SOURCE_DIR}/java/jni.inc)
```

Comment:
1. Conditional include would avoid file processing alltogether

1. A separate project would inherit the settings from the main file. Check the gtest out. There all is done is add_library. Either way you do not have to change much.